test/k8s/datapath_configuration.go

```diff
@@ -585,7 +586,167 @@ var _ = Describe("K8sDatapathConfig", func() {
+				res1, cancel1, err := kubectl.ExecInHostNetNSInBackground(context.TODO(), k8s1NodeName, cmd)
+				ExpectWithOffset(2, err).Should(BeNil(), "Cannot exec tcpdump in bg")
+				res2, cancel2, err := kubectl.ExecInHostNetNSInBackground(context.TODO(), k8s2NodeName, cmd)
+				ExpectWithOffset(2, err).Should(BeNil(), "Cannot exec tcpdump in bg")
+
+				// HTTP connectivity test (pod2pod)
 
+				cmdRes, err := kubectl.ExecUntilMatch(randomNamespace, srcPod, helpers.CurlTimeout("http://%s/", time.Second, net.JoinHostPort(dstService, "80")), dstPod)
+				if shouldBeSuccessful {
+					cmdRes.ExpectSuccess("Failed to curl svc on dst pod")
+					ExpectWithOffset(2, err).Should(BeNil(), "Failed to curl svc on dst pod")
+				} else {
+					ExpectWithOffset(2, err).ShouldNot(BeNil(), "Should not be able to curl svc on dst pod")
+				}
+
+				// Check that no unencrypted pod2pod traffic was captured on the direct routing device
+				cancel1()
+				cancel2()
+				ExpectWithOffset(2, res1.CombineOutput().String()).Should(Not(ContainSubstring("1 packet captured")))
+				ExpectWithOffset(2, res2.CombineOutput().String()).Should(Not(ContainSubstring("1 packet captured")))
+			}
+
+			checkNoLeakP2P(srcPod, srcPodIP.String(), dstPodIP.String(), false)
+			checkNoLeakP2RemoteService(srcPod, dstPod, srcPodIP.String(), dsSvcIP.String(), false)
+
+			// Check that the src pod can reach the remote host
+			kubectl.ExecPodCmd(randomNamespace, srcPod, helpers.Ping(k8s2IP)).
+				ExpectSuccess("Failed to ping k8s2 host from src pod")
+
+			// Check that the remote host can reach the dst pod
+			kubectl.ExecInHostNetNS(context.TODO(), k8s1NodeName,
+				helpers.CurlFail("http://%s:80/", dstPodIP)).ExpectSuccess("Failed to curl dst pod from k8s1")
+
+			// Re-enable operator so that all endpoints are eventually updated
+			kubectl.SetCiliumOperatorReplicas(2)
+
+			// Due to IPCache update delays, it can take up to a few seconds
+			// before both nodes have added the new pod IPs to their allowedIPs
+			// list, which can cause flakes in CI. Therefore wait for the
+			// IPs to be present on both nodes before performing the test
+			waitForAllowedIP := func(ciliumPod, ip string) {
+				jsonpath := fmt.Sprintf(`{.encryption.wireguard.interfaces[*].peers[*].allowed-ips[?(@=='%s')]}`, ip)
+				ciliumCmd := fmt.Sprintf(`cilium debuginfo --output jsonpath="%s"`, jsonpath)
+				expected := fmt.Sprintf("jsonpath=%s", ip)
+				err := kubectl.CiliumExecUntilMatch(ciliumPod, ciliumCmd, expected)
+				Expect(err).To(BeNil(), "ip %q not in allowedIPs of pod %q", ip, ciliumPod)
+			}
+
+			waitForAllowedIP(ciliumPodK8s1, fmt.Sprintf("%s/32", dstPodIP))
+			waitForAllowedIP(ciliumPodK8s2, fmt.Sprintf("%s/32", srcPodIP))
+
+			checkNoLeakP2P(srcPod, srcPodIP.String(), dstPodIP.String(), true)
+
+			checkNoLeakP2RemoteService(srcPod, dstPod, srcPodIP.String(), dsSvcIP.String(), true)
+		}
+
+		It("Pod2pod is encrypted in tunneling mode with per-endpoint routes STRICT MODE vxlan", func() {
```

Comment:
Just for consistency with the below change. We are already in the `Wireguard encryption strict mode` Context so no need to tell here again that we are in strict mode.
```suggestion
		It("Pod-to-pod traffic is encrypted in tunneling mode with per-endpoint routes", func() {
```